### DISCUSSION TITLE
Projection of FE solution on RB space

### DISCUSSION THREAD

**Discussion by niravshah241** [RBniCS/RBniCSx#7] [General]
I have a problem similar to tutorial 03_hole with scheme similar to following:

```
problem = ProblemOnDeformedDomain()
solution = problem.solve(mu_solve) # mu_solve is the parameter numpy array
reduced_problem = PODGInefficientReducedProblem(problem)
# Compute modes using POD (rbnicsx.backends.proper_orthogonal_decomposition)
reduced_problem.basis_functions.extend(modes)
```

I would like to know whether there is method for projecting the FE "solution" on the reduced basis space such as reduced_problem.project .

**Comment by francesco-ballarin** [comment]
You should be able to mimic what is done in the reduced solve, no? It is enough to precompute the reduced matrix associated to the L^2 inner product (very much like matrices associated to the lhs operator are precomputed) and compute on the fly the reduced vector associated to the L^2 product between the FE solution and the reduced basis. Afterwards, you can solve that system using petsc4py, as done in the reduced solve.
